### PR TITLE
Add transformers id to hub requests

### PR DESCRIPTION
**Pull request by philschmid**
# What does this PR do?

This PR adds a `TRANSFORMERS_ID` const, which helps us to group the several files request against the hub.

### CODE REVIEW

File: src/transformers/file_utils.py [hunk 1]
```diff
@@ -213,6 +214,7 @@
 PYTORCH_PRETRAINED_BERT_CACHE = os.getenv("PYTORCH_PRETRAINED_BERT_CACHE", default_cache_path)
 PYTORCH_TRANSFORMERS_CACHE = os.getenv("PYTORCH_TRANSFORMERS_CACHE", PYTORCH_PRETRAINED_BERT_CACHE)
 TRANSFORMERS_CACHE = os.getenv("TRANSFORMERS_CACHE", PYTORCH_TRANSFORMERS_CACHE)
+TRANSFORMERS_ID = uuid4().hex
```
Comment (n1t0) [inline src/transformers/file_utils.py:217]: I had a preference for the previous name `SESSION_ID` that I find more informative.

**Review by n1t0 — APPROVED** [review]
LGTM! I added a small nitpick below, but nothing important.

**Review by sgugger — APPROVED** [review]
I defer to @n1t0 in this case :-)

File: src/transformers/file_utils.py [hunk 2]
```diff
@@ -1211,7 +1213,7 @@ def http_user_agent(user_agent: Union[Dict, str, None] = None) -> str:
     """
     Formats a user-agent string with basic info about a request.
     """
-    ua = "transformers/{}; python/{}".format(__version__, sys.version.split()[0])
+    ua = f"transformers/{__version__}; python/{sys.version.split()[0]}; session_id/{SESSION_ID}"
```
Comment (julien-c) [inline src/transformers/file_utils.py:1216]: can we trim the id slightly to e.g. 12 chars? @philschmid cc @n1t0

Comment (n1t0) [reply src/transformers/file_utils.py:1216]: I don't have a strong opinion on this

Comment (philschmid) [reply src/transformers/file_utils.py:1216]: I wanted to leave this unchanged and adding 20 more characters to it won`t be that big of a problem. But I am open to adjusting it.

**Pull request merged**